File: base.database/src/main/java/alvin/base/database/dbflow/domain/models/Person.java
package alvin.adv.database.dbflow.domain.models;

import com.raizlabs.android.dbflow.annotation.Column;
import com.raizlabs.android.dbflow.annotation.PrimaryKey;
import com.raizlabs.android.dbflow.annotation.Table;
import com.raizlabs.android.dbflow.structure.BaseModel;

import java.time.LocalDate;

import alvin.adv.database.dbflow.domain.FlowDatabase;
import alvin.adv.database.common.domain.models.Gender;
import alvin.adv.database.common.domain.models.IPerson;
import alvin.lib.common.dbflow.converts.LocalDateConvert;

@Table(database = FlowDatabase.class, name = "user")
public class Person extends BaseModel implements IPerson {

    @PrimaryKey(autoincrement = true)
    private int id;

    @Column
    private String name;

    @Column
    private Gender gender;

    @Column(typeConverter = LocalDateConvert.class)
    private LocalDate birthday;

    Person() {
    }

    public Person(String name, Gender gender, LocalDate birthday) {
        this.name = name;
        this.gender = gender;
        this.birthday = birthday;
    }

    @Override
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    @Override
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Override
    public Gender getGender() {
        return gender;
    }

    public void setGender(Gender gender) {
        this.gender = gender;
    }

    @Override
    public LocalDate getBirthday() {
        return birthday;
    }

    public void setBirthday(LocalDate birthday) {
        this.birthday = birthday;
    }
}

File: base.kotlin/src/main/java/alvin/base/kotlin/dbflow/domain/models/Person.kt
package alvin.base.kotlin.dbflow.domain.models

import alvin.base.kotlin.dbflow.domain.MainDatabase
import alvin.base.kotlin.dbflow.domain.converts.LocalDateConvert
import com.raizlabs.android.dbflow.annotation.Column
import com.raizlabs.android.dbflow.annotation.PrimaryKey
import com.raizlabs.android.dbflow.annotation.Table
import java.time.LocalDate

@Table(name = "person", database = MainDatabase::class)
data class Person(
        @PrimaryKey(autoincrement = true)
        var id: Int = 0,

        @Column(name = "name")
        var name: String? = null,

        @Column(name = "gender")
        var gender: Gender? = null,

        @Column(name = "birthday", typeConverter = LocalDateConvert::class)
        var birthday: LocalDate? = null
) {
    constructor(
            name: String?,
            gender: Gender?,
            birthday: LocalDate?
    ) : this(0, name, gender, birthday)
}